src/HtmlUtils.tsx

```diff
@@ -548,6 +551,12 @@ export function bodyToHtml(content: IContent, highlights: Optional<string[]>, op
                 // by an attempt to search for 'foobar'.  Then again, the search query probably wouldn't work either
                 // XXX: hacky bodge to temporarily apply a textFilter to the sanitizeParams structure.
                 sanitizeParams.textFilter = function (safeText) {
+                    if (opts.emotes) {
+                        return highlighter
+                            .applyHighlights(safeText, safeHighlights!)
+                            .join("")
+                            .replace(CUSTOM_EMOTES_REGEX, (m) => (opts?.emotes?.has(m) ? opts.emotes.get(m)! : m));
+                    }
```

Comment:
Note for @matrix-org/product - this rendering is done on the receiving side, regardless of whether a sender intended it. It may not play well with bridges. If a room has a `:short:` emote defined in its state then a user will not be able to send the literal `:short:` without it being inflated into the emote image.